From 862ee76acbf83a2d785809734b02c1929c4a0859 Mon Sep 17 00:00:00 2001
From: Matthias Maier <tamiko@43-1.org>
Date: Wed, 14 Jun 2017 03:22:11 -0500
Subject: [PATCH] Add IS_IN guard to multiarch/IFUNC implementations

The two preceding rtld patches  add the first reference to strcspn to
ld.so, which pulls in the multiarch/IFUNC implementations because the
files in question lack an IS_IN (libc) guard (which essentially disables
IFUNCs in ld.so).

Patch by Florian Weimer <fweimer@redhat.com>
---
 sysdeps/i386/i686/multiarch/strcspn-c.c | 6 ++++--
 sysdeps/i386/i686/multiarch/varshift.c  | 4 +++-
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/sysdeps/i386/i686/multiarch/strcspn-c.c b/sysdeps/i386/i686/multiarch/strcspn-c.c
index 6d61e19..ec230fb 100644
--- a/sysdeps/i386/i686/multiarch/strcspn-c.c
+++ b/sysdeps/i386/i686/multiarch/strcspn-c.c
@@ -1,2 +1,4 @@
-#define __strcspn_sse2 __strcspn_ia32
-#include <sysdeps/x86_64/multiarch/strcspn-c.c>
+#if IS_IN (libc)
+# define __strcspn_sse2 __strcspn_ia32
+# include <sysdeps/x86_64/multiarch/strcspn-c.c>
+#endif
diff --git a/sysdeps/i386/i686/multiarch/varshift.c b/sysdeps/i386/i686/multiarch/varshift.c
index 7760b96..6742a35 100644
--- a/sysdeps/i386/i686/multiarch/varshift.c
+++ b/sysdeps/i386/i686/multiarch/varshift.c
@@ -1 +1,3 @@
-#include <sysdeps/x86_64/multiarch/varshift.c>
+#if IS_IN (libc)
+# include <sysdeps/x86_64/multiarch/varshift.c>
+#endif
-- 
2.13.0

